pkg/status/controller.go

```diff
@@ -204,6 +204,14 @@ func (c *Controller) handleObject(obj any) {
 	c.enqueueObject(obj)
 }
 
+// handleObjectDeletion is like handleObject except that handleObjectDeletion calls
+// enqueueObjectDeletion instead of enqueueObject
+func (c *Controller) handleObjectDeletion(obj any) {
+	rObj := obj.(runtime.Object)
+	c.logger.V(4).Info("Got object deletion event", "obj", util.RefToRuntimeObj(rObj))
+	c.enqueueObjectDeletion(obj)
```

Comment:
The approach of enqueuing an object rather than an object reference can go wrong. The workqueue suppresses duplicates, as identified by Go's primitive `==`. The object is not `==` its https://github.com/kubernetes/client-go/blob/v0.28.4/tools/cache/object-names.go#L24 . The FIFO part of the workqueue is ordered, but retries spend some time in the delay part before going back in the FIFO. So the following sequence of events can get handled incorrectly.

1. User creates object.
2. Controller is notified of create and enqueues ObjectName.
3. Controller processes ObjectName, seeing new object.
4. User deletes object.
5. Controller is notified of delete and enqueues whole object.
6. User re-creates object.
7. Controller is notified of create and enqueues ObjectName.
8. Controller processes whole old object but gets an error, puts whole object back for retry later.
9. Controller processes ObjectName, seeing new version of object. Does right stuff?
10. Controller processes whole old object, undoes stuff just done.
11. Nothing else happens, user is sad.